tests/types.py

```diff
@@ -0,0 +1,13 @@
+from typing import Any, Awaitable, Callable, Dict, Iterable
+
+from starlette.responses import Response
+from starlette.testclient import TestClient
+
+TestClientFactory = Callable[..., TestClient]
```

Comment:
Can this be a protocol?

Reply:
I have made the changes but I had to change the naming to `ClientFactoryProtocol` because using `TestClientFactory` leads to another issue.

If i use the name starting with `Test` pytest is trying to collect it as if it were a test case or test class, but it is not intended to be either.

I tried two things for this:
1. add `# pytest: no collect` but this did not fix the issue
2. add `__test__ = False` but this led to some other errors:
`All protocol members must have explicitly declared types`
`Incompatible return value type (got "partial[TestClient]", expected "TestClientFactory")  [return-value]` 
`"partial" is missing following "TestClientFactory" protocol member` inside `conftest.py`

Is that naming ok or should I try another fix or keep it without protocol?